homeassistant/components/caldav/calendar.py

```diff
@@ -199,21 +199,19 @@ def update(self):
             dt.start_of_local_day(), dt.start_of_local_day() + timedelta(days=1)
         )
 
-        # Change start/end dates of recurring events to today if they occur today.
+        # Change start/end dates of the next recurring event to today if they occur after now.
         # For recurring events, some servers return the original event with recurrence rules
         # and they would not be properly parsed using their original start/end dates.
         for event in results:
             vevent = event.instance.vevent
             rrules = vevent.getrruleset()
             if rrules:
-                sld = dt.start_of_local_day()
-                if sld.date() in (rrule.date() for rrule in rrules):
-                    vevent.dtstart.value = vevent.dtstart.value.replace(
-                        year=sld.year, month=sld.month, day=sld.day
-                    )
-                    vevent.dtend.value = vevent.dtend.value.replace(
-                        year=sld.year, month=sld.month, day=sld.day
-                    )
+                rrule = next((rrule for rrule in rrules if dt.now() < rrule), None)
```

Comment:
Now that I think about it, this won't work for ongoing events: events that have already started but are not over yet.  I'll have to fix that.